packages/botonic-react/src/webchat/cover-component/index.tsx

```diff
@@ -0,0 +1,33 @@
+import React, { useContext } from 'react'
+
+import { WEBCHAT } from '../../constants'
+import { WebchatContext } from '../../contexts'
+
+interface Props {
+  component: any
+  componentProps: any
+}
+
+export const CoverComponent = ({ component, componentProps }: Props) => {
+  const { getThemeProperty, toggleCoverComponent, webchatState } =
+    useContext(WebchatContext)
+
+  const Cover = component
+
+  const coverComponentProps = getThemeProperty(
+    WEBCHAT.CUSTOM_PROPERTIES.coverComponentProps,
+    componentProps
+  )
+
+  const closeCoverComponent = () => {
+    toggleCoverComponent(false)
+  }
+
+  if (!Cover) {
+    return null
+  }
+
+  return Cover && webchatState.isCoverComponentOpen ? (
```

Comment:
no need to check for `Cover` again if already checked above.

just return `webchatState.isCoverComponentOpen && <Cover ...`
and before `if (!Cover || !webchatState.isCoverComponentOpen`

Reply:
applied